### PR TITLE
transaction deferable

### PR DESCRIPTION
**Pull request by insani7y**

### CODE REVIEW

File: src/driver/transaction.rs [hunk 1]
```diff
@@ -38,6 +39,7 @@ impl RustTransaction {
         rollback_savepoint: Arc<tokio::sync::RwLock<HashSet<String>>>,
         isolation_level: Option<IsolationLevel>,
         read_variant: Option<ReadVariant>,
+        deferable: Option<bool>,
```
Comment (chandr-andr) [inline src/driver/transaction.rs:42]: You need to add this parameter in `pyi` file for correct autosuggestions and type hints. You can find these fields in `python/psqlpy/_internal`

File: src/driver/transaction.rs [hunk 2]
```diff
@@ -118,6 +122,10 @@ impl RustTransaction {
             querystring.push_str(format!(" {}", &read_var.to_str_option()).as_str());
         }
 
+        if let Some(_) = self.deferable {
```
Comment (chandr-andr) [inline src/driver/transaction.rs:125]: I think it's better to use the default `if condition`, `if let` usually used in other situations:

```
if self.deferable.is_some() {
    querystring.push_str("SET CONSTRAINTS ALL DEFERRED");
}
```

File: python/psqlpy/_internal/__init__.pyi [hunk 3]
```diff
@@ -1,12 +1,12 @@
 from enum import Enum
-from typing import Any, Dict, List, Optional
```
Comment (chandr-andr) [inline python/psqlpy/_internal/__init__.pyi:2]: You need to return this. In python version <3.9 there is no default support for `list`, `dict` and etc.

Comment (insani7y) [reply python/psqlpy/_internal/__init__.pyi:2]: Can't you just 
from __future__ import annotations ?

Comment (insani7y) [reply python/psqlpy/_internal/__init__.pyi:2]: Also it is not needed, cause it's a stub file and type checkers always treat those files with that import statement

**Comment by chandr-andr** [comment]
Need to add new functionality to Readme section

**Comment by chandr-andr** [comment]
@insani7y Could you please add tests for transaction creation parameters?  
I mean for isolation, read variant and deferable.

**Pull request closed without merging**